yaml: fix properties syntax to 'mapping' instead of 'series'

erwango opened the pull request:
According to yaml syntaxic rules, 'properties' described in dts
bindings yaml files could be seen as 'mapping'(key/value couple),
instead of 'series' (list of single elements).
Using mapping, yaml mappings will be converted by yaml python
library as python 'dict' which will ease treatment
(instead of current list as were before this commit).
Same treatment is applied to 'inherits'.

script extract_dts_inlcude is updated to take change of yaml_list
structre into account. This allows some code simplification. Largest
impact is yaml_collapse function which works now allow complete
overload method on all the attributes of a yaml nodes.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
recheck

erwango commented:
Failed test is passed at my end.
Checking python version

mnkp commented:
Shouldn't we also update (remove the leading hyphen) all the .yaml files in samples/ and tests/ directories?

erwango commented:
@mnkp: Initial purpose was to fix this to ease yaml related parsing in extract_dts_include script.
Other yaml files are parsed by other scripts (maybe not all python btw).

So this would be different jobs, with impacts specific to each script. I guess this is up to each script owner to check this. But maybe we could raise an issue.

ydamigos commented:
Could you also update device_node.yaml.template?

ydamigos reviewed (commented):
Just a minor comment, otherwise it looks fine.

erwango commented:
@mnkp: Submitted #5109

erwango commented:
@ydamigos : Was done already but it missed one bit. Should be ok now

ydamigos commented on dts/bindings/device_node.yaml.template:
```diff
@@ -12,27 +12,27 @@ description: >
 
 # reg is used to denote mmio registers
-  - reg:
+      reg:
```
Could you align it with compatible and interrupts?

ydamigos reviewed (approved):
LGTM. Just a small alignment.

erwango committed:
yaml: fix properties syntax to 'mapping' instead of 'series'

According to yaml syntaxic rules, 'properties' described in dts
bindings yaml files could be seen as 'mapping'(key/value couple),
instead of 'series' (list of single elements).
yaml 'mappings' will then be converted by yaml python library as
python 'dict' which will ease treatment (instead of current list
as were before this commit).
Same treatment is applied to 'inherits'.

script extract_dts_inlcude is updated to take change of yaml_list
structre into account. This allows some code simplification. Largest
impact is yaml_collapse function which works now allow complete
overload method on all the attributes of a yaml nodes.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

dbkinder reviewed (commented):
I don't see any doc changes to review here...
